client/jetpack-connect/signup.js

```diff
@@ -212,7 +262,128 @@ export class JetpackSignup extends Component {
 			</LoggedOutFormLinks>
 		);
 	}
+
+	renderWooDnaFooter( footerLinks ) {
+		const { authQuery } = this.props;
+		footerLinks.push(
+			<LoggedOutFormLinkItem key="back" href={ authQuery.redirectAfterAuth }>
+				<Gridicon size={ 18 } icon="arrow-left" />{ ' ' }
+				{
+					// translators: eg: Return to The WordPress.com Blog
+					this.props.translate( 'Return to %(sitename)s', {
+						args: { sitename: decodeEntities( authQuery.blogname ) },
+					} )
+				}
+			</LoggedOutFormLinkItem>
+		);
+		return <LoggedOutFormLinks>{ footerLinks }</LoggedOutFormLinks>;
+	}
+
+	renderWooDna() {
+		const { authQuery, isFullLoginFormVisible, translate, usernameOrEmail } = this.props;
+		const {
+			isCreatingAccount,
+			signUpUsernameOrEmail,
+			loginSocialConnect,
+			loginTwoFactorAuthType,
+		} = this.state;
+		let header, subHeader, content;
+		const footerLinks = [];
+		const email = signUpUsernameOrEmail || usernameOrEmail || authQuery.userEmail;
+		const wooDna = this.getWooDnaConfig();
+		let pageTitle;
+
+		if ( 'login' === this.state.wooDnaFormType ) {
+			if ( isFullLoginFormVisible ) {
+				header = translate( 'Log in to your WordPress.com account' );
+				/* translators: pluginName is the name of the Woo extension that initiated the connection flow */
+				subHeader = translate(
+					'Your account will enable you to start using the features and benefits offered by %(pluginName)s',
+					{
+						args: {
+							pluginName: wooDna.name( translate ),
+						},
+					}
+				);
+				pageTitle = translate( 'Login to WordPress.com' );
+				footerLinks.push(
+					<LoggedOutFormLinkItem key="signup" onClick={ this.showWooDnaSignupView }>
+						{ this.props.translate( 'Create a new account' ) }
+					</LoggedOutFormLinkItem>
+				);
+				footerLinks.push(
+					<LoggedOutFormLinkItem
+						key="lostpassword"
+						href={ addQueryArgs(
+							{ action: 'lostpassword' },
+							login( { locale: this.props.locale } )
+						) }
+					>
+						{ this.props.translate( 'Lost your password?' ) }
+					</LoggedOutFormLinkItem>
+				);
+			} else {
+				header = wooDna.name( translate );
+				subHeader = translate( 'Enter your email address to get started' );
+				pageTitle = translate( 'Connect' );
+			}
+			content = (
+				<LoginBlock
+					locale={ this.props.locale }
+					footer={ this.renderWooDnaFooter( footerLinks ) }
+					userEmail={ email }
+					socialConnect={ loginSocialConnect }
+					twoFactorAuthType={ loginTwoFactorAuthType }
+					onTwoFactorRequested={ ( authType ) =>
+						this.setState( { loginTwoFactorAuthType: authType } )
+					}
+					onSocialConnectStart={ () => this.setState( { loginSocialConnect: true } ) }
+				/>
+			);
+		} else {
+			// Woo DNA sign-up form
+			header = wooDna.name( translate );
+			subHeader = translate( 'Create an account' );
+			pageTitle = translate( 'Create a WordPress.com account' );
+			footerLinks.push(
+				<LoggedOutFormLinkItem key="login" onClick={ () => this.showWooDnaLoginView() }>
```

Comment:
No, because then `showWooDnaLoginView` would be called with `event` as the first parameter, and [that would be wrong](https://github.com/Automattic/wp-calypso/blob/1aebc79bc32e3de59174e88c633964a58dfe1c6f/client/jetpack-connect/signup.js#L121).